### PR TITLE
Validate modules arg spec fixes

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
This PR makes updates to `validate-modules` so that the `--arg-spec` argument can be supplied.

With this PR, all modules can be run through with `--arg-spec`.

We still don't run with `--arg-spec` by default, because there are many (82) issues uncovered by this additional check.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/test/sanity/validate-modules

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by sivel** [comment]
Something of note here, is that if a module cannot be executed to get it's argspec, that doesn't reflect on any exit code, it is just ignored currently, although a traceback will be printed.

**Comment by gundalow** [comment]
> Something of note here, is that if a module cannot be executed to get it's argspec, that doesn't reflect on any exit code, it is just ignored currently, although a traceback will be printed.

I think that's OK.

**Comment by sivel** [comment]
@mattclay this has been rebased. There was some change made that is causing `azure_rm_storageaccount.py` to fail again, and I wasn't able to figure it out. So it's printing an error, and being skipped by validation.  I think in that case, we'll need to fix the module, instead of worrying with more hacks.

The printed error does not affect the execution as mentioned above, it's printed for info, but skipped.

**Review by mattclay — APPROVED** [review]

**Comment by sivel** [comment]
The PR has been updated to make a failure to retrieve argument spec due to errors on import and error.

Once https://github.com/ansible/ansible/pull/34757 is merged, I'll extend this PR to enable the `--arg-spec` flag and add ignores for the current errors.

File: test/runner/lib/sanity/validate_modules.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test(self, args, targets):
         cmd = [
             'python%s' % args.python_version,
             'test/sanity/validate-modules/validate-modules',
-            '--format', 'json',
+            '--format', 'json', '--arg-spec'
```
Comment (mattclay) [inline test/runner/lib/sanity/validate_modules.py]: I'd prefer adding this on a line by itself with a trailing comma.

File: test/sanity/validate-modules/ignore.txt [hunk 2]
```diff
@@ -0,0 +1,129 @@
+lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py E317
```
Comment (mattclay) [inline test/sanity/validate-modules/ignore.txt:1]: The lines should be unique, no need for duplicates.

Comment (sivel) [reply test/sanity/validate-modules/ignore.txt:1]: Thanks, it didn't even occur to me that there may have been duplicates when I wrote out that file.

**Comment by sivel** [comment]
@mattclay comments addressed

**Review by mattclay — APPROVED** [review]

**Comment by mattclay** [comment]
The only CI failure was unrelated to this PR.

**Pull request merged**